### PR TITLE
Enable async prefetching for ReadOptions.readahead_size

### PR DESCRIPTION
**Pull request by akankshamahajan15**
Summary: Currently async prefetching is enabled for implicit internal auto readahead in FilePrefetchBuffer if `ReadOptions.async_io` is set. This PR enables async prefetching for `ReadOptions.readahead_size` when `ReadOptions.async_io` is set true.

Test Plan: Update unit test

### CODE REVIEW

File: table/block_based/partitioned_index_reader.cc [hunk 1]
```diff
@@ -82,6 +82,8 @@ InternalIteratorBase<IndexValue>* PartitionIndexReader::NewIterator(
     ro.io_timeout = read_options.io_timeout;
     ro.adaptive_readahead = read_options.adaptive_readahead;
     ro.async_io = read_options.async_io;
+    ro.readahead_size = read_options.readahead_size;
```
Comment (akankshamahajan15) [inline table/block_based/partitioned_index_reader.cc:85]: I enabled readahead_size for index blocks also.

Comment (anand1976) [reply table/block_based/partitioned_index_reader.cc:85]: I don't know if this is desirable. The index size is much smaller than data, and enabling explicit readahead_size would significantly increase the memory footprint of the iterator, especially if readahead_size is a large value.

Comment (akankshamahajan15) [reply table/block_based/partitioned_index_reader.cc:85]: Ok. I will remove it then.

**Review by anand1976 — APPROVED** [review]
LGTM

**Pull request closed without merging**